### PR TITLE
ui: Disable tip for sampled queries

### PR DESCRIPTION
**Pull request by zhangvi7**
If query is already sampled, we hide the user tip

### CODE REVIEW

File: querybook/webapp/components/QueryExecution/SamplingToolTip.tsx [hunk 1]
```diff
@@ -12,17 +12,19 @@ interface SamplingTooltipProps {
...
     queryExecution: { status, id },
     onSamplingInfoClick,
     hasSamplingTables,
+    sampleRate,
 }) => {
     const { enabled, sampling_tool_tip_delay: delay } =
         PublicConfig.table_sampling;
 
-    const queryCanBeSampled = enabled && hasSamplingTables;
+    const queryCanBeSampled = enabled && hasSamplingTables && sampleRate <= 0;
```
Comment (jczhong84) [inline querybook/webapp/components/QueryExecution/SamplingToolTip.tsx:27]: nit: the name `queryCanBeSampled` here seems inappropriate anymore

Comment (zhangvi7) [reply querybook/webapp/components/QueryExecution/SamplingToolTip.tsx:27]: @jczhong84 how about `isQuerySamplingEligible`?

**Comment by czgu** [comment]
hey @jczhong84 and @zhangvi7 do you know why is sampling rate info piped down from composer to the execution? isn't it available via query meta? https://github.com/pinterest/querybook/pull/1465/files

File: querybook/webapp/components/QueryExecution/SamplingToolTip.tsx (hunk 1 above)
Comment (jczhong84) [reply querybook/webapp/components/QueryExecution/SamplingToolTip.tsx:27]: or just remove the variable here but check them directly inside the useEffect
```
if(enabled && hasSamplingTables && sampleRate <= 0 && status <= QueryExecutionStatus.RUNNING`)
```

**Comment by zhangvi7** [comment]
@czgu Its used the determine if the currently written query, before executing, is eligible for sampling based on tables and then decide we to show user tip

**Comment by zhangvi7** [comment]
Included bug fix for query composer:
- Previously if tables in query change, we did not update samplingTables to include updated sampleRate.
- And execution metadata was not updated properly if there are no sampling tables

Added getter functions to provide updated values for sampleRate and samplingTables, similar to DataDocQueryCell

File: querybook/webapp/components/QueryRunButton/QueryRunButton.tsx [hunk 2]
```diff
@@ -105,15 +105,16 @@ export const QueryRunButton = React.forwardRef<
...
-            !disabled && TABLE_SAMPLING_CONFIG.enabled && hasSamplingTables ? (
-                <TableSamplingSelector
-                    sampleRate={sampleRate}
-                    setSampleRate={onSampleRateChange}
-                    tooltipPos={runButtonTooltipPos}
-                    onTableSamplingInfoClick={onTableSamplingInfoClick}
-                />
-            ) : null;
+        const tableSamplingDOM = (
+            // !disabled && TABLE_SAMPLING_CONFIG.enabled && hasSamplingTables ? (
```
Comment (jczhong84) [inline querybook/webapp/components/QueryRunButton/QueryRunButton.tsx]: can we remove those commented out lines if they are not needed

File: querybook/webapp/components/QueryComposer/QueryComposer.tsx [hunk 3]
```diff
@@ -507,9 +499,34 @@ const QueryComposer: React.FC = () => {
...
+        () => (Object.keys(samplingTables).length > 0 ? sampleRate : -1),
+        [sampleRate, samplingTables]
+    );
+
+    const getSamplingTables = useCallback(() => {
+        Object.keys(samplingTables).forEach((tableName) => {
+            samplingTables[tableName].sample_rate = getSampleRate();
+        });
+        return samplingTables;
+    }, [getSampleRate, samplingTables]);
```
Comment (jczhong84) [inline querybook/webapp/components/QueryComposer/QueryComposer.tsx:512]: can these be accommodated into the  `useTableSampleRate`?

File: querybook/webapp/components/QueryExecution/SamplingToolTip.tsx [hunk 4]
```diff
@@ -38,7 +43,7 @@ export const SamplingTooltip: React.FC<SamplingTooltipProps> = ({
                 setShowSamplingTip(false);
             };
         }
-    }, [status, id, queryCanBeSampled, delay]);
+    }, [id, delay, enabled, hasSamplingTables, sampleRate, status]);
```
Comment (jczhong84) [inline querybook/webapp/components/QueryExecution/SamplingToolTip.tsx:46]: what is this id?

Comment (zhangvi7) [reply querybook/webapp/components/QueryExecution/SamplingToolTip.tsx:46]: <id> here is the current query_execution_id, I added it here so the effect reexecute when we run new query

File: querybook/webapp/components/QueryComposer/QueryComposer.tsx [hunk 5]
```diff
@@ -178,7 +182,19 @@ const useTableSampleRate = (dispatch: Dispatch, environmentId: number) => {
...
+
+    return { sampleRate, setSampleRate, getSampleRate, getSamplingTables };
```
Comment (jczhong84) [inline querybook/webapp/components/QueryComposer/QueryComposer.tsx:197]: with `getSampleRate`, do we still need `sampleRate`?

Comment (zhangvi7) [reply querybook/webapp/components/QueryComposer/QueryComposer.tsx]: removed!

**Review by jczhong84 — APPROVED** [review]

**Pull request merged**